src/metabase/lib/fe_util.cljc

```diff
@@ -77,6 +77,24 @@
     (expand-temporal-expression expression-clause)
     expression-clause))
 
+(defn- expandable-case-or-if-expression?
+  [[op _options & _args]]
+  (or (= op :case)
+      (= op :if)))
+
+(defn- expand-case-or-if-expression
+  [[op options & args]]
+  (let [clauses (into [] (apply concat (first args)))
+        fallback (second args)
+        res (conj clauses fallback)]
```

Comment:
You can destructure on possibly-missing parts of the incoming clause; they'll be `nil` if missing, so there's no need for `& args` + `first` and `second`.

The 3-arg form of `into` takes `(into target transducer input-seq)`. Transducers are a big topic! But for our purposes here, `cat` is a transducer that expects a list-of-lists and concats them all. (One level deep.)

`concat` returns a lazy sequence, not a vector. Since we want to return a clause as a vector, the final return should use `(into [...] ...)` rather than `concat`.

```suggestion
  [[op options clause-pairs fallback]]
  (let [clauses (into [] cat clause-pairs)]
    (into [op options] (conj clauses fallback))))
```